internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -220,25 +224,80 @@ func (h *PolicyChangeHandler) handleFleetServerHosts(ctx context.Context, c *con
 	return nil
 }
 
-func clientEqual(k1 remote.Config, k2 remote.Config) bool {
-	if k1.Protocol != k2.Protocol {
+// applyConfigWithPrecedence applies Proxy and TLS configs, but ignores empty or
+// absent ones.
+// That way a proxy or TLS config set during install/enroll using cli flags
+// won't be overridden by an absent or empty proxy from fleet-server.
+// However, if there is a proxy or TLS config sent by fleet-server, it'll take
+// precedence. Therefore, it's not possible to remove a proxy or TLS config once
+// it's set.
+func (h *PolicyChangeHandler) applyConfigWithPrecedence(cfg remote.Config) {
+	defaultcfg := configuration.DefaultFleetAgentConfig()
+
+	if cfg.Protocol != defaultcfg.Client.Protocol ||
+		cfg.Host != defaultcfg.Client.Host ||
+		!slices.Equal(cfg.Hosts, defaultcfg.Client.Hosts) ||
+		cfg.Path != defaultcfg.Client.Path {
+		h.config.Fleet.Client.Protocol = cfg.Protocol
+		h.config.Fleet.Client.Path = cfg.Path
+		h.config.Fleet.Client.Host = cfg.Host
+		h.config.Fleet.Client.Hosts = cfg.Hosts
+	}
+
+	if cfg.Transport.Proxy.URL == nil ||
+		cfg.Transport.Proxy.URL.String() == "" {
+		h.log.Debug("proxy from fleet is empty or null, the proxy will not be changed")
+	} else {
+		h.config.Fleet.Client.Transport.Proxy = cfg.Transport.Proxy
+		h.log.Debug("received proxy from fleet, applying it")
+	}
+
+	emptyCertificate := tlscommon.CertificateConfig{}
+	if cfg.Transport.TLS != nil {
+		if h.config.Fleet.Client.Transport.TLS == nil {
+			h.config.Fleet.Client.Transport.TLS = &tlscommon.Config{}
+		}
+
+		// client certificate
+		if cfg.Transport.TLS.Certificate == emptyCertificate {
+			h.log.Debug("TLS certificate/key from fleet are empty or null, the TLS config will not be changed")
+		} else {
+			h.config.Fleet.Client.Transport.TLS.Certificate = cfg.Transport.TLS.Certificate
```

Comment:
Is the Certificate configuration from Fleet always complete? Does it or could it ever allow updating only individual fields? 

If it does this unintentionally clears the ones that weren't set, for example if someone  sets only the `Certificate` but not the `Key` and `Key` were set previously.

I suppose if this happened unintentionally and broke things it would be reverted, but it would be hard for a user to debug. 

We either need to make debugging that failure easier or just prevent it from happening entirely.